vm/collector.hpp

```diff
@@ -97,18 +100,12 @@ template <typename TargetGeneration, typename Policy> struct collector {
   void trace_partial_objects(cell start, cell card_start, cell card_end) {
     object* obj = (object*)start;
     cell end = start + obj->binary_payload_start();
-    if (card_start < end) {
-      start += sizeof(cell);
-
-      start = std::max(start, card_start);
-      end = std::min(end, card_end);
+    start += sizeof(cell);
```

Comment:
No, but if `card_start >= end` then it must be that `start >= end` and then `visit_object_array` won't visit anything.